pandas/plotting/_matplotlib/core.py

```diff
@@ -226,13 +224,25 @@ def _validate_color_args(self):
                 styles = [self.style]
             # need only a single match
             for s in styles:
-                for char in s:
-                    if char in matplotlib.colors.BASE_COLORS:
-                        raise ValueError(
-                            "Cannot pass 'style' string with a color symbol and "
-                            "'color' keyword argument. Please use one or the other or "
-                            "pass 'style' without a color symbol"
-                        )
+                if self._color_in_style(s):
+                    raise ValueError(
+                        "Cannot pass 'style' string with a color symbol and "
+                        "'color' keyword argument. Please use one or the "
+                        "other or pass 'style' without a color symbol"
+                    )
+
+    @staticmethod
+    def _color_in_style(style):
+        """
+        Is there a color letter in the style string?
+
+        Returns
+        -------
+        bool
+        """
+        from matplotlib.colors import BASE_COLORS
+
+        return not set(BASE_COLORS).isdisjoint(style)
```

Comment:
Is it necessary to add a new stastic method (and 10+ lines of code) for a one-liner? IMO just replacing the guard with `set(BASE_COLORS).isdisjoint(style)` should be enough, with the new import replacing the old one

Reply:
@joooeey on a second thought, this is probably alright here, as it means you you don't have to duplicate later in `_apply_style_colors`.

Can you add type annotations? I think `style` is a string here, so
```python
_color_in_style(style: str) -> bool:
```

Reply:
Also, the short summary in the docstring should start with an infinitive verb, see https://pandas.pydata.org/docs/development/contributing_docstring.html

Reply:
Working on it! With type annotations it looks way cleaner than with the lengthy docstring which is a little silly in a private method. This would be the first type annotation in `core.py`, yey!

Reply:
:laughing: yeah, they're encouraged for new development, though old code doesn't necessarily have them